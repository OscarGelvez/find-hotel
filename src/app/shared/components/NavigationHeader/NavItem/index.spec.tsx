import '@testing-library/jest-dom';

import { cleanup, render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { NavItem } from './index';

afterEach(cleanup);

it('rederizar el componente NavBrand sin errores', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <NavItem label={''} to={''} />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renderiza el componente NavItem cuando se envía información', () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <NavItem label={'Reservas'} to={'/reservas'} />
    </BrowserRouter>
  );
  expect(getByTestId('nav-item')).toContainElement(
    getByTestId('link-nav-item')
  );
});

it('Compara snapshot del componente NavItem cuando se envía información', () => {
  const element = renderer
    .create(
      <BrowserRouter>
        <NavItem label={'Reservas'} to={'/reservas'} />
      </BrowserRouter>
    )
    .toJSON();
  expect(element).toMatchSnapshot();
});
