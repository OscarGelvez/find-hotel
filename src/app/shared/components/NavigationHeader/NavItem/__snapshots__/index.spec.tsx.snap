// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compara snapshot del componente NavItem cuando se envía información 1`] = `
<div
  className="sc-AxirZ jxvKur"
  data-testid="nav-item"
>
  <a
    className="sc-AxjAm jqjyPd"
    data-testid="link-nav-item"
    href="/reservas"
    onClick={[Function]}
  >
    Reservas
  </a>
</div>
`;
